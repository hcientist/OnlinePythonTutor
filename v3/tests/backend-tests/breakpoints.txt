# when '#break' is the last string in a line, then stop at only those lines

x = [1, 2, 3]
y = [4, 5, 6]
z = y
y = x
x = z

x = [1, 2, 3] # a different [1, 2, 3] list!
y = x
x.append(4) #break
y.append(5)
z = [1, 2, 3, 4, 5] # a different list!  #break
x.append(6)
y.append(7)
y = "hello" #break


def foo(lst): #break
    lst.append("hello") #break
    bar(lst)

def bar(myLst):
    print(myLst) #break

foo(x)
foo(z)
