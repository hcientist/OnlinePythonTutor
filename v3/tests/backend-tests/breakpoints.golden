{
  "code": "# when '#break' is the last string in a line, then stop at only those lines\n\nx = [1, 2, 3]\ny = [4, 5, 6]\nz = y\ny = x\nx = z\n\nx = [1, 2, 3] # a different [1, 2, 3] list!\ny = x\nx.append(4) #break\ny.append(5)\nz = [1, 2, 3, 4, 5] # a different list!  #break\nx.append(6)\ny.append(7)\ny = \"hello\" #break\n\n\ndef foo(lst): #break\n    lst.append(\"hello\") #break\n    bar(lst)\n\ndef bar(myLst):\n    print(myLst) #break\n\nfoo(x)\nfoo(z)\n", 
  "trace": [
    {
      "ordered_globals": [
        "x", 
        "y", 
        "z"
      ], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "y": [
          "REF", 
          3
        ], 
        "x": [
          "REF", 
          3
        ], 
        "z": [
          "REF", 
          2
        ]
      }, 
      "heap": {
        "2": [
          "LIST", 
          4, 
          5, 
          6
        ], 
        "3": [
          "LIST", 
          1, 
          2, 
          3
        ]
      }, 
      "line": 11, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "x", 
        "y", 
        "z"
      ], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "y": [
          "REF", 
          3
        ], 
        "x": [
          "REF", 
          3
        ], 
        "z": [
          "REF", 
          2
        ]
      }, 
      "heap": {
        "2": [
          "LIST", 
          4, 
          5, 
          6
        ], 
        "3": [
          "LIST", 
          1, 
          2, 
          3, 
          4
        ]
      }, 
      "line": 12, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "x", 
        "y", 
        "z"
      ], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "y": [
          "REF", 
          3
        ], 
        "x": [
          "REF", 
          3
        ], 
        "z": [
          "REF", 
          2
        ]
      }, 
      "heap": {
        "2": [
          "LIST", 
          4, 
          5, 
          6
        ], 
        "3": [
          "LIST", 
          1, 
          2, 
          3, 
          4, 
          5
        ]
      }, 
      "line": 13, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "x", 
        "y", 
        "z"
      ], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "y": [
          "REF", 
          3
        ], 
        "x": [
          "REF", 
          3
        ], 
        "z": [
          "REF", 
          1
        ]
      }, 
      "heap": {
        "1": [
          "LIST", 
          1, 
          2, 
          3, 
          4, 
          5
        ], 
        "3": [
          "LIST", 
          1, 
          2, 
          3, 
          4, 
          5
        ]
      }, 
      "line": 14, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "x", 
        "y", 
        "z"
      ], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "y": [
          "REF", 
          3
        ], 
        "x": [
          "REF", 
          3
        ], 
        "z": [
          "REF", 
          1
        ]
      }, 
      "heap": {
        "1": [
          "LIST", 
          1, 
          2, 
          3, 
          4, 
          5
        ], 
        "3": [
          "LIST", 
          1, 
          2, 
          3, 
          4, 
          5, 
          6, 
          7
        ]
      }, 
      "line": 16, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "x", 
        "y", 
        "z"
      ], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "y": "hello", 
        "x": [
          "REF", 
          3
        ], 
        "z": [
          "REF", 
          1
        ]
      }, 
      "heap": {
        "1": [
          "LIST", 
          1, 
          2, 
          3, 
          4, 
          5
        ], 
        "3": [
          "LIST", 
          1, 
          2, 
          3, 
          4, 
          5, 
          6, 
          7
        ]
      }, 
      "line": 19, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "x", 
        "y", 
        "z", 
        "foo"
      ], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "y": "hello", 
        "x": [
          "REF", 
          3
        ], 
        "foo": [
          "REF", 
          4
        ], 
        "z": [
          "REF", 
          1
        ]
      }, 
      "heap": {
        "1": [
          "LIST", 
          1, 
          2, 
          3, 
          4, 
          5
        ], 
        "3": [
          "LIST", 
          1, 
          2, 
          3, 
          4, 
          5, 
          6, 
          7
        ], 
        "4": [
          "FUNCTION", 
          "foo(lst)", 
          null
        ]
      }, 
      "line": 23, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "x", 
        "y", 
        "z", 
        "foo", 
        "bar"
      ], 
      "stdout": "", 
      "func_name": "foo", 
      "stack_to_render": [
        {
          "frame_id": 1, 
          "encoded_locals": {
            "lst": [
              "REF", 
              3
            ]
          }, 
          "is_highlighted": true, 
          "is_parent": false, 
          "func_name": "foo", 
          "is_zombie": false, 
          "parent_frame_id_list": [], 
          "unique_hash": "foo_f1", 
          "ordered_varnames": [
            "lst"
          ]
        }
      ], 
      "globals": {
        "y": "hello", 
        "x": [
          "REF", 
          3
        ], 
        "foo": [
          "REF", 
          4
        ], 
        "bar": [
          "REF", 
          5
        ], 
        "z": [
          "REF", 
          1
        ]
      }, 
      "heap": {
        "1": [
          "LIST", 
          1, 
          2, 
          3, 
          4, 
          5
        ], 
        "3": [
          "LIST", 
          1, 
          2, 
          3, 
          4, 
          5, 
          6, 
          7
        ], 
        "4": [
          "FUNCTION", 
          "foo(lst)", 
          null
        ], 
        "5": [
          "FUNCTION", 
          "bar(myLst)", 
          null
        ]
      }, 
      "line": 19, 
      "event": "call"
    }, 
    {
      "ordered_globals": [
        "x", 
        "y", 
        "z", 
        "foo", 
        "bar"
      ], 
      "stdout": "", 
      "func_name": "foo", 
      "stack_to_render": [
        {
          "frame_id": 1, 
          "encoded_locals": {
            "lst": [
              "REF", 
              3
            ]
          }, 
          "is_highlighted": true, 
          "is_parent": false, 
          "func_name": "foo", 
          "is_zombie": false, 
          "parent_frame_id_list": [], 
          "unique_hash": "foo_f1", 
          "ordered_varnames": [
            "lst"
          ]
        }
      ], 
      "globals": {
        "y": "hello", 
        "x": [
          "REF", 
          3
        ], 
        "foo": [
          "REF", 
          4
        ], 
        "bar": [
          "REF", 
          5
        ], 
        "z": [
          "REF", 
          1
        ]
      }, 
      "heap": {
        "1": [
          "LIST", 
          1, 
          2, 
          3, 
          4, 
          5
        ], 
        "3": [
          "LIST", 
          1, 
          2, 
          3, 
          4, 
          5, 
          6, 
          7
        ], 
        "4": [
          "FUNCTION", 
          "foo(lst)", 
          null
        ], 
        "5": [
          "FUNCTION", 
          "bar(myLst)", 
          null
        ]
      }, 
      "line": 20, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "x", 
        "y", 
        "z", 
        "foo", 
        "bar"
      ], 
      "stdout": "", 
      "func_name": "foo", 
      "stack_to_render": [
        {
          "frame_id": 1, 
          "encoded_locals": {
            "lst": [
              "REF", 
              3
            ]
          }, 
          "is_highlighted": true, 
          "is_parent": false, 
          "func_name": "foo", 
          "is_zombie": false, 
          "parent_frame_id_list": [], 
          "unique_hash": "foo_f1", 
          "ordered_varnames": [
            "lst"
          ]
        }
      ], 
      "globals": {
        "y": "hello", 
        "x": [
          "REF", 
          3
        ], 
        "foo": [
          "REF", 
          4
        ], 
        "bar": [
          "REF", 
          5
        ], 
        "z": [
          "REF", 
          1
        ]
      }, 
      "heap": {
        "1": [
          "LIST", 
          1, 
          2, 
          3, 
          4, 
          5
        ], 
        "3": [
          "LIST", 
          1, 
          2, 
          3, 
          4, 
          5, 
          6, 
          7, 
          "hello"
        ], 
        "4": [
          "FUNCTION", 
          "foo(lst)", 
          null
        ], 
        "5": [
          "FUNCTION", 
          "bar(myLst)", 
          null
        ]
      }, 
      "line": 21, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "x", 
        "y", 
        "z", 
        "foo", 
        "bar"
      ], 
      "stdout": "", 
      "func_name": "bar", 
      "stack_to_render": [
        {
          "frame_id": 1, 
          "encoded_locals": {
            "lst": [
              "REF", 
              3
            ]
          }, 
          "is_highlighted": false, 
          "is_parent": false, 
          "func_name": "foo", 
          "is_zombie": false, 
          "parent_frame_id_list": [], 
          "unique_hash": "foo_f1", 
          "ordered_varnames": [
            "lst"
          ]
        }, 
        {
          "frame_id": 2, 
          "encoded_locals": {
            "myLst": [
              "REF", 
              3
            ]
          }, 
          "is_highlighted": true, 
          "is_parent": false, 
          "func_name": "bar", 
          "is_zombie": false, 
          "parent_frame_id_list": [], 
          "unique_hash": "bar_f2", 
          "ordered_varnames": [
            "myLst"
          ]
        }
      ], 
      "globals": {
        "y": "hello", 
        "x": [
          "REF", 
          3
        ], 
        "foo": [
          "REF", 
          4
        ], 
        "bar": [
          "REF", 
          5
        ], 
        "z": [
          "REF", 
          1
        ]
      }, 
      "heap": {
        "1": [
          "LIST", 
          1, 
          2, 
          3, 
          4, 
          5
        ], 
        "3": [
          "LIST", 
          1, 
          2, 
          3, 
          4, 
          5, 
          6, 
          7, 
          "hello"
        ], 
        "4": [
          "FUNCTION", 
          "foo(lst)", 
          null
        ], 
        "5": [
          "FUNCTION", 
          "bar(myLst)", 
          null
        ]
      }, 
      "line": 24, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "x", 
        "y", 
        "z", 
        "foo", 
        "bar"
      ], 
      "stdout": "[1, 2, 3, 4, 5, 6, 7, 'hello']\n", 
      "func_name": "bar", 
      "stack_to_render": [
        {
          "frame_id": 1, 
          "encoded_locals": {
            "lst": [
              "REF", 
              3
            ]
          }, 
          "is_highlighted": false, 
          "is_parent": false, 
          "func_name": "foo", 
          "is_zombie": false, 
          "parent_frame_id_list": [], 
          "unique_hash": "foo_f1", 
          "ordered_varnames": [
            "lst"
          ]
        }, 
        {
          "frame_id": 2, 
          "encoded_locals": {
            "__return__": null, 
            "myLst": [
              "REF", 
              3
            ]
          }, 
          "is_highlighted": true, 
          "is_parent": false, 
          "func_name": "bar", 
          "is_zombie": false, 
          "parent_frame_id_list": [], 
          "unique_hash": "bar_f2", 
          "ordered_varnames": [
            "myLst", 
            "__return__"
          ]
        }
      ], 
      "globals": {
        "y": "hello", 
        "x": [
          "REF", 
          3
        ], 
        "foo": [
          "REF", 
          4
        ], 
        "bar": [
          "REF", 
          5
        ], 
        "z": [
          "REF", 
          1
        ]
      }, 
      "heap": {
        "1": [
          "LIST", 
          1, 
          2, 
          3, 
          4, 
          5
        ], 
        "3": [
          "LIST", 
          1, 
          2, 
          3, 
          4, 
          5, 
          6, 
          7, 
          "hello"
        ], 
        "4": [
          "FUNCTION", 
          "foo(lst)", 
          null
        ], 
        "5": [
          "FUNCTION", 
          "bar(myLst)", 
          null
        ]
      }, 
      "line": 24, 
      "event": "return"
    }, 
    {
      "ordered_globals": [
        "x", 
        "y", 
        "z", 
        "foo", 
        "bar"
      ], 
      "stdout": "[1, 2, 3, 4, 5, 6, 7, 'hello']\n", 
      "func_name": "foo", 
      "stack_to_render": [
        {
          "frame_id": 1, 
          "encoded_locals": {
            "__return__": null, 
            "lst": [
              "REF", 
              3
            ]
          }, 
          "is_highlighted": true, 
          "is_parent": false, 
          "func_name": "foo", 
          "is_zombie": false, 
          "parent_frame_id_list": [], 
          "unique_hash": "foo_f1", 
          "ordered_varnames": [
            "lst", 
            "__return__"
          ]
        }
      ], 
      "globals": {
        "y": "hello", 
        "x": [
          "REF", 
          3
        ], 
        "foo": [
          "REF", 
          4
        ], 
        "bar": [
          "REF", 
          5
        ], 
        "z": [
          "REF", 
          1
        ]
      }, 
      "heap": {
        "1": [
          "LIST", 
          1, 
          2, 
          3, 
          4, 
          5
        ], 
        "3": [
          "LIST", 
          1, 
          2, 
          3, 
          4, 
          5, 
          6, 
          7, 
          "hello"
        ], 
        "4": [
          "FUNCTION", 
          "foo(lst)", 
          null
        ], 
        "5": [
          "FUNCTION", 
          "bar(myLst)", 
          null
        ]
      }, 
      "line": 21, 
      "event": "return"
    }, 
    {
      "ordered_globals": [
        "x", 
        "y", 
        "z", 
        "foo", 
        "bar"
      ], 
      "stdout": "[1, 2, 3, 4, 5, 6, 7, 'hello']\n", 
      "func_name": "foo", 
      "stack_to_render": [
        {
          "frame_id": 3, 
          "encoded_locals": {
            "lst": [
              "REF", 
              1
            ]
          }, 
          "is_highlighted": true, 
          "is_parent": false, 
          "func_name": "foo", 
          "is_zombie": false, 
          "parent_frame_id_list": [], 
          "unique_hash": "foo_f3", 
          "ordered_varnames": [
            "lst"
          ]
        }
      ], 
      "globals": {
        "y": "hello", 
        "x": [
          "REF", 
          3
        ], 
        "foo": [
          "REF", 
          4
        ], 
        "bar": [
          "REF", 
          5
        ], 
        "z": [
          "REF", 
          1
        ]
      }, 
      "heap": {
        "1": [
          "LIST", 
          1, 
          2, 
          3, 
          4, 
          5
        ], 
        "3": [
          "LIST", 
          1, 
          2, 
          3, 
          4, 
          5, 
          6, 
          7, 
          "hello"
        ], 
        "4": [
          "FUNCTION", 
          "foo(lst)", 
          null
        ], 
        "5": [
          "FUNCTION", 
          "bar(myLst)", 
          null
        ]
      }, 
      "line": 19, 
      "event": "call"
    }, 
    {
      "ordered_globals": [
        "x", 
        "y", 
        "z", 
        "foo", 
        "bar"
      ], 
      "stdout": "[1, 2, 3, 4, 5, 6, 7, 'hello']\n", 
      "func_name": "foo", 
      "stack_to_render": [
        {
          "frame_id": 3, 
          "encoded_locals": {
            "lst": [
              "REF", 
              1
            ]
          }, 
          "is_highlighted": true, 
          "is_parent": false, 
          "func_name": "foo", 
          "is_zombie": false, 
          "parent_frame_id_list": [], 
          "unique_hash": "foo_f3", 
          "ordered_varnames": [
            "lst"
          ]
        }
      ], 
      "globals": {
        "y": "hello", 
        "x": [
          "REF", 
          3
        ], 
        "foo": [
          "REF", 
          4
        ], 
        "bar": [
          "REF", 
          5
        ], 
        "z": [
          "REF", 
          1
        ]
      }, 
      "heap": {
        "1": [
          "LIST", 
          1, 
          2, 
          3, 
          4, 
          5
        ], 
        "3": [
          "LIST", 
          1, 
          2, 
          3, 
          4, 
          5, 
          6, 
          7, 
          "hello"
        ], 
        "4": [
          "FUNCTION", 
          "foo(lst)", 
          null
        ], 
        "5": [
          "FUNCTION", 
          "bar(myLst)", 
          null
        ]
      }, 
      "line": 20, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "x", 
        "y", 
        "z", 
        "foo", 
        "bar"
      ], 
      "stdout": "[1, 2, 3, 4, 5, 6, 7, 'hello']\n", 
      "func_name": "foo", 
      "stack_to_render": [
        {
          "frame_id": 3, 
          "encoded_locals": {
            "lst": [
              "REF", 
              1
            ]
          }, 
          "is_highlighted": true, 
          "is_parent": false, 
          "func_name": "foo", 
          "is_zombie": false, 
          "parent_frame_id_list": [], 
          "unique_hash": "foo_f3", 
          "ordered_varnames": [
            "lst"
          ]
        }
      ], 
      "globals": {
        "y": "hello", 
        "x": [
          "REF", 
          3
        ], 
        "foo": [
          "REF", 
          4
        ], 
        "bar": [
          "REF", 
          5
        ], 
        "z": [
          "REF", 
          1
        ]
      }, 
      "heap": {
        "1": [
          "LIST", 
          1, 
          2, 
          3, 
          4, 
          5, 
          "hello"
        ], 
        "3": [
          "LIST", 
          1, 
          2, 
          3, 
          4, 
          5, 
          6, 
          7, 
          "hello"
        ], 
        "4": [
          "FUNCTION", 
          "foo(lst)", 
          null
        ], 
        "5": [
          "FUNCTION", 
          "bar(myLst)", 
          null
        ]
      }, 
      "line": 21, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "x", 
        "y", 
        "z", 
        "foo", 
        "bar"
      ], 
      "stdout": "[1, 2, 3, 4, 5, 6, 7, 'hello']\n", 
      "func_name": "bar", 
      "stack_to_render": [
        {
          "frame_id": 3, 
          "encoded_locals": {
            "lst": [
              "REF", 
              1
            ]
          }, 
          "is_highlighted": false, 
          "is_parent": false, 
          "func_name": "foo", 
          "is_zombie": false, 
          "parent_frame_id_list": [], 
          "unique_hash": "foo_f3", 
          "ordered_varnames": [
            "lst"
          ]
        }, 
        {
          "frame_id": 4, 
          "encoded_locals": {
            "myLst": [
              "REF", 
              1
            ]
          }, 
          "is_highlighted": true, 
          "is_parent": false, 
          "func_name": "bar", 
          "is_zombie": false, 
          "parent_frame_id_list": [], 
          "unique_hash": "bar_f4", 
          "ordered_varnames": [
            "myLst"
          ]
        }
      ], 
      "globals": {
        "y": "hello", 
        "x": [
          "REF", 
          3
        ], 
        "foo": [
          "REF", 
          4
        ], 
        "bar": [
          "REF", 
          5
        ], 
        "z": [
          "REF", 
          1
        ]
      }, 
      "heap": {
        "1": [
          "LIST", 
          1, 
          2, 
          3, 
          4, 
          5, 
          "hello"
        ], 
        "3": [
          "LIST", 
          1, 
          2, 
          3, 
          4, 
          5, 
          6, 
          7, 
          "hello"
        ], 
        "4": [
          "FUNCTION", 
          "foo(lst)", 
          null
        ], 
        "5": [
          "FUNCTION", 
          "bar(myLst)", 
          null
        ]
      }, 
      "line": 24, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "x", 
        "y", 
        "z", 
        "foo", 
        "bar"
      ], 
      "stdout": "[1, 2, 3, 4, 5, 6, 7, 'hello']\n[1, 2, 3, 4, 5, 'hello']\n", 
      "func_name": "bar", 
      "stack_to_render": [
        {
          "frame_id": 3, 
          "encoded_locals": {
            "lst": [
              "REF", 
              1
            ]
          }, 
          "is_highlighted": false, 
          "is_parent": false, 
          "func_name": "foo", 
          "is_zombie": false, 
          "parent_frame_id_list": [], 
          "unique_hash": "foo_f3", 
          "ordered_varnames": [
            "lst"
          ]
        }, 
        {
          "frame_id": 4, 
          "encoded_locals": {
            "__return__": null, 
            "myLst": [
              "REF", 
              1
            ]
          }, 
          "is_highlighted": true, 
          "is_parent": false, 
          "func_name": "bar", 
          "is_zombie": false, 
          "parent_frame_id_list": [], 
          "unique_hash": "bar_f4", 
          "ordered_varnames": [
            "myLst", 
            "__return__"
          ]
        }
      ], 
      "globals": {
        "y": "hello", 
        "x": [
          "REF", 
          3
        ], 
        "foo": [
          "REF", 
          4
        ], 
        "bar": [
          "REF", 
          5
        ], 
        "z": [
          "REF", 
          1
        ]
      }, 
      "heap": {
        "1": [
          "LIST", 
          1, 
          2, 
          3, 
          4, 
          5, 
          "hello"
        ], 
        "3": [
          "LIST", 
          1, 
          2, 
          3, 
          4, 
          5, 
          6, 
          7, 
          "hello"
        ], 
        "4": [
          "FUNCTION", 
          "foo(lst)", 
          null
        ], 
        "5": [
          "FUNCTION", 
          "bar(myLst)", 
          null
        ]
      }, 
      "line": 24, 
      "event": "return"
    }, 
    {
      "ordered_globals": [
        "x", 
        "y", 
        "z", 
        "foo", 
        "bar"
      ], 
      "stdout": "[1, 2, 3, 4, 5, 6, 7, 'hello']\n[1, 2, 3, 4, 5, 'hello']\n", 
      "func_name": "foo", 
      "stack_to_render": [
        {
          "frame_id": 3, 
          "encoded_locals": {
            "__return__": null, 
            "lst": [
              "REF", 
              1
            ]
          }, 
          "is_highlighted": true, 
          "is_parent": false, 
          "func_name": "foo", 
          "is_zombie": false, 
          "parent_frame_id_list": [], 
          "unique_hash": "foo_f3", 
          "ordered_varnames": [
            "lst", 
            "__return__"
          ]
        }
      ], 
      "globals": {
        "y": "hello", 
        "x": [
          "REF", 
          3
        ], 
        "foo": [
          "REF", 
          4
        ], 
        "bar": [
          "REF", 
          5
        ], 
        "z": [
          "REF", 
          1
        ]
      }, 
      "heap": {
        "1": [
          "LIST", 
          1, 
          2, 
          3, 
          4, 
          5, 
          "hello"
        ], 
        "3": [
          "LIST", 
          1, 
          2, 
          3, 
          4, 
          5, 
          6, 
          7, 
          "hello"
        ], 
        "4": [
          "FUNCTION", 
          "foo(lst)", 
          null
        ], 
        "5": [
          "FUNCTION", 
          "bar(myLst)", 
          null
        ]
      }, 
      "line": 21, 
      "event": "return"
    }
  ]
}
